tmail-backend/webadmin/webadmin-oidc-backchannel/src/main/java/com/linagora/tmail/webadmin/OidcBackchannelLogoutRoutes.java

```diff
@@ -0,0 +1,127 @@
+    private static final String APPLICATION_FORM_URLENCODED_VALUE = "application/x-www-form-urlencoded";
+    private static final String TOKEN_PARAM = "logout_token";
+    private static final String SID_PROPERTY = "sid";
+
+    private final OidcTokenCache oidcTokenCache;
+    private final ObjectMapper objectMapper;
+    private final JsonTransformer jsonTransformer;
+
+    @Inject
+    public OidcBackchannelLogoutRoutes(OidcTokenCache oidcTokenCache, JsonTransformer jsonTransformer) {
+        this.oidcTokenCache = oidcTokenCache;
+        this.jsonTransformer = jsonTransformer;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    @Override
+    public String getBasePath() {
+        return BASE_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.post(BASE_PATH, addRevokedToken(), jsonTransformer);
+    }
+
+    public Route addRevokedToken() {
+        return (request, response) -> {
+            if (!StringUtils.startsWith(request.contentType(), APPLICATION_FORM_URLENCODED_VALUE)) {
+                response.status(HttpStatus.UNSUPPORTED_MEDIA_TYPE_415);
+                return "Unsupported Content-Type";
+            }
+
+            String token = request.queryParams(TOKEN_PARAM);
+            Preconditions.checkArgument(StringUtils.isNotEmpty(token), "Missing logout token");
+
+            try {
+                extractSidFromLogoutToken(token)
+                    .ifPresentOrElse(sid -> {
+                        LOGGER.debug("Add new revoked token has sid: " + sid);
+                        oidcTokenCache.invalidate(sid).block();
+                    }, () -> LOGGER.warn("Cannot extract sid from logout token: {}", token));
+                response.status(HttpStatus.OK_200);
+                return Constants.EMPTY_BODY;
+            } catch (Exception e) {
+                throw ErrorResponder.builder()
+                    .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+                    .type(ErrorResponder.ErrorType.SERVER_ERROR)
+                    .message(String.format("Error while adding revoked token '%s'", token))
+                    .cause(e)
+                    .haltError();
+            }
+        };
+    }
+
+    private Optional<Sid> extractSidFromLogoutToken(String token) {
+        try {
+            List<String> parts = Splitter.on('.')
+                .trimResults()
+                .omitEmptyStrings()
+                .splitToList(token);
+            if (parts.size() < 2) {
+                return Optional.empty();
+            }
+
+            String payloadJson = new String(BaseEncoding.base64Url().decode(parts.get(1)), StandardCharsets.UTF_8);
+            Map<String, Object> payloadMap = objectMapper.readValue(payloadJson, new TypeReference<>() {
+            });
+
+            return Optional.ofNullable((String) payloadMap.getOrDefault(SID_PROPERTY, null))
+                .map(Sid::new);
```

Comment:
This won't work with opaque token, no?

IMO we should resolve token -> sid using `OidcTokenCache.associatedInformation`.

Reply:
logout_token is different than the oidc token I believe